packages/react-devtools-extensions/src/background.js

```diff
@@ -51,7 +51,9 @@ chrome.runtime.onConnect.addListener(function(port) {
   ports[tab][name] = port;
 
   if (ports[tab].devtools && ports[tab]['content-script']) {
-    doublePipe(ports[tab].devtools, ports[tab]['content-script']);
+    doublePipe(ports[tab].devtools, ports[tab]['content-script'], tab);
+    // clean up so that we can rebuild the double pipe if the page is reloaded
+    ports[tab] = null;
```

Comment:
Should this be moved to `shutdown`? On line 176,  `ports[id]` is accessed. (although I don't know what it does)

Reply:
Good catch. Fixed.